src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -0,0 +1,171 @@
+                {
+                    logDirectory = Path.GetDirectoryName(nativeLogFile);
+                }
+
+                if (logDirectory == null)
+                {
+                    if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+                    {
+                        var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                        logDirectory = windowsDefaultDirectory;
+                    }
+                    else
+                    {
+                        // Linux
+                        logDirectory = NixDefaultDirectory;
+                    }
+                }
+
+                logDirectory = CreateDirectoryIfMissing(logDirectory) ?? Path.GetTempPath();
+            }
+            catch
+            {
+                // The try block may throw a SecurityException if not granted the System.Security.Permissions.FileIOPermission
+                // because of the following API calls
+                //   - Directory.Exists
+                //   - Environment.GetFolderPath
+                //   - Path.GetTempPath
+
+                // Unsafe to log
+                logDirectory = null;
+            }
+
+            return logDirectory;
+        }
+
+        private static string CreateDirectoryIfMissing(string pathToCreate)
+        {
+            try
+            {
+                Directory.CreateDirectory(pathToCreate);
+                return pathToCreate;
+            }
+            catch
+            {
+                // Unable to create the directory meaning that the user will have to create it on their own.
+                // It is unsafe to log here, so return null to defer deciding what the path is
+                return null;
+            }
+        }
+
+        private static string SetStartupLogFilePath()
+        {
+            if (LogDirectory == null)
+            {
+                return null;
+            }
+
+            try
+            {
+                var process = Process.GetCurrentProcess();
+                // Do our best to not block other processes on write
+                return Path.Combine(LogDirectory, $"dotnet-tracer-loader-{process.ProcessName}-{process.Id}.log");
+            }
+            catch
+            {
+                // We can't get the process info
+                return Path.Combine(LogDirectory, "dotnet-tracer-loader.log");
+            }
+        }
+
+        private static bool IsDebugEnabled()
+        {
+            try
+            {
+                var ddTraceDebugValue = Environment.GetEnvironmentVariable("DD_TRACE_DEBUG");
+
+                if (ddTraceDebugValue == null)
+                {
+                    return false;
+                }
+
+                switch (ddTraceDebugValue.ToUpperInvariant())
+                {
+                    case "TRUE":
+                    case "YES":
+                    case "T":
```

Comment:
Not something to change now, but I'm surprised that in the original extension method we allow `T` for `TRUE`, but not `Y` for `YES`.